jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jsonp/JsonpDiagnosticCollector.java

```diff
@@ -32,7 +42,39 @@ public void completeDiagnostic(Diagnostic diagnostic) {
 
     @Override
     public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnostics) {
+        if (unit == null) {
+            return;
+        }
+        List<MethodInvocation> allMethodInvocations = ASTUtils.getMethodInvocations(unit);
+        List<MethodInvocation> createPointerInvocations = allMethodInvocations.stream()
+                .filter(this::isCreatePointerInvocation).collect(Collectors.toList());
+        for (MethodInvocation m: createPointerInvocations) {
+            Expression arg = (Expression) m.arguments().get(0);
+            if (isInvalidArgument(arg)) {
+                try {
+                    Range range = JDTUtils.toRange(unit, arg.getStartPosition(), arg.getLength());
+                    Diagnostic diagnostic = new Diagnostic(range, JsonpConstants.CREATE_POINTER_ERROR_MESSAGE);
```

Comment:
Consider a comment for the added logic here detailing what this diagnostic covers/does

Reply:
Done